--- /usr/bin/dailystrips.debian	2011-07-27 16:28:26.000000000 +0200
+++ /usr/bin/dailystrips	2018-03-26 17:49:55.011374926 +0200
@@ -531,7 +531,7 @@
 	if ($options{'stripnav'}) {
 		print "<font face=\"helvetica\">Strips:</font><br>\n";
 		for (@strips) {
-			my ($strip, $name) = (split(/;/, $_))[0,1];
+			my ($strip, $name) = $$_[0,1];
 			print "<a href=\"#$strip\">$name</A>&nbsp;&nbsp;";
 		}
 		print "\n<br><br>";
@@ -550,12 +550,12 @@
 }
 
 for (@strips) {
-	my ($strip, $name, $homepage, $img_addr, $referer, $prefetch, $artist) = split(/;/, $_);
+	my ($strip, $name, $homepage, $img_addr, $referer, $prefetch, $artist, $alt) = @$_;
 	my ($img_line, $local_name, $local_name_dir, $local_name_file, $local_name_ext, $image, $ext,
 	   $local_name_yesterday, $local_name_yesterday_dir, $local_name_yesterday_file, $local_name_yesterday_ext);
 	
 	if ($options{'verbose'} and $options{'local'}) {
-		warn "Downloading strip file for " . lc((split(/;/, $_))[0]) . "\n";
+		warn "Downloading strip file for " . lc($strip) . "\n";
 	}
 	
 	if ($img_addr =~ "^unavail") {
@@ -791,6 +791,11 @@
 	</tr>
 	<tr>
 		<td>
+			$alt
+		</td>
+	</tr>
+	<tr>
+		<td>
 			$img_line
 			<p>&nbsp;</p>
 		</td>
@@ -896,7 +901,7 @@
 
 sub get_strip {
 	my ($strip) = @_;
-	my ($page, $addr);
+	my ($page, $addr, $alt, @row);
 	
 	if ($options{'date'} and $defs{$strip}{'provides'} eq "latest") {
 		if ($options{'verbose'}) {
@@ -940,6 +945,13 @@
 					$addr = $defs{$strip}{'baseurl'} . $match . $defs{$strip}{'urlsuffix'};
 				}
 			}
+
+                        if ($defs{$strip}{'altpattern'}) {
+                            $page =~ /$defs{$strip}{'altpattern'}/si;
+                            $alt = ${1};
+                        } else {
+                            $alt = "";
+                        }
 		}
 		
 	} elsif ($defs{$strip}{'type'} eq "generate") {
@@ -947,8 +959,9 @@
 	}
 	
 	unless ($addr =~ /^(https?:\/\/|unavail)/io) { $addr = "http://" . $addr }
-	
-	push(@strips,"$strip;$defs{$strip}{'name'};$defs{$strip}{'homepage'};$addr;$defs{$strip}{'referer'};$defs{$strip}{'prefetch'};$defs{$strip}{'artist'}");
+
+        @row = ($strip, $defs{$strip}{'name'}, $defs{$strip}{'homepage'}, $addr, $defs{$strip}{'referer'}, $defs{$strip}{'prefetch'}, $defs{$strip}{'artist'}, $alt);
+	push(@strips, \@row);
 }
 
 sub get_defs {
@@ -1230,6 +1243,10 @@
 			{
 				$defs{$strip}{'searchpattern'} = $1;
 			}
+			elsif (/^altpattern\s+(.+)$/i)
+			{
+				$defs{$strip}{'altpattern'} = $1;
+			}
 			elsif (/^matchpart\s+(.+)$/i)
 			{
 				unless ($1 =~ /^(\d+)$/) {
