--- /home/wosc/git-notifier-0.6/github-notifier	2015-01-08 16:53:18.000000000 +0100
+++ /usr/local/gitnotify/github-notifier	2018-01-31 08:25:07.000000000 +0100
@@ -1,4 +1,4 @@
-#! /usr/bin/env python
+#!/usr/local/gitnotify/bin/python
 #
 # Needs http://jacquev6.github.io/PyGithub.
 
@@ -58,7 +58,7 @@
         error("cannot run command '%s': %s" % (args, e))
 
     if child.returncode != 0:
-        error("command failed with exit code %d" % child.returncode)
+        error("command '%s' failed with exit code %d" % (args, child.returncode))  # wosc patched
 
 def gitClone(repo):
     assert not os.path.exists(repo.path)
